src/js/components/widgets-link/widget-link-number.react.js

```diff
@@ -0,0 +1,287 @@
+        this.ctx.beginPath();
+        this.ctx.moveTo(this.width * 0.5, 0);
+        this.ctx.lineTo(this.width * 0.5, this.height);
+        this.ctx.closePath();
+        this.ctx.stroke();
+
+        // Triangle line
+        this.ctx.fillStyle = this.overPoint ? this.selColor : this.fnColor;
+        this.ctx.beginPath();
+        this.ctx.moveTo(this.width * 0.5, 5);
+        this.ctx.lineTo(this.width * 0.48, 0);
+        this.ctx.lineTo(this.width * 0.52, 0);
+        this.ctx.closePath();
+        this.ctx.fill();
+
+        let times = 3;
+        let unit = 40;
+        let step = this.width / unit;
+        let sections = unit * times;
+
+        let offsetX = this.offsetX;
+
+        if (Math.abs(this.offsetX - this.width * 0.5) > this.width * 0.5) {
+            offsetX = (this.offsetX - this.width * 0.5) % (this.width * 0.5) + this.width;
+        }
+
+        this.ctx.strokeStyle = this.dimColor;
+        this.ctx.beginPath();
+        for (let i = 0; i < sections; i++) {
+            let l = (i % (unit / 2) === 0) ? this.height * 0.35 : (i % (unit / 4) === 0) ? this.height * 0.2 : this.height * 0.1;
+            this.ctx.moveTo(i * step - offsetX, this.height * 0.5 - l);
+            this.ctx.lineTo(i * step - offsetX, this.height * 0.5 + l);
+        }
+        this.ctx.stroke();
+
+        let val = Math.round(((this.value - this.min) / this.range) * this.width);
+
+        // point
+        this.ctx.strokeStyle = this.overPoint ? this.selColor : this.fnColor;
+        this.ctx.lineWidth = 1;
+        this.ctx.beginPath();
+        this.ctx.moveTo(this.offsetX + val, this.height * 0.5);
+        this.ctx.lineTo(this.offsetX + val, this.height);
+        this.ctx.closePath();
+        this.ctx.stroke();
+
+        this.overPoint = false;
+    }
+
+    /**
+     * Set the value of the number
+     *
+     * @param value - new value to set our number to
+     */
+    _setValue (value) {
+        this.value = value;
+        this.offsetX = this.value * this.center;
+    }
+
+    /**
+     * Update CodeMirror
+     *
+     * @param string - the new number to write out to CodeMirror
+     */
+    _updateEditor (string) {
+        let start = { line: this.cursor.line, ch: this.match.start };
+        let end = { line: this.cursor.line, ch: this.match.end };
+        this.match.end = this.match.start + string.length;
+        editor.replaceRange(string, start, end);
+    }
+
+    /* Mouse, scroll and click commands */
+    /* These event callbacks are provided by React */
+
+    /**
+     * Widget links are handled slightly differently. For now they are simply unmounted from the DOM and recreated again
+     * Meaning, handleClick will only be called once to unmount the widget
+     */
+    handleClick () {
+        this.setState({ displayPicker: !this.state.displayPicker });
+
+        let widgetlink = document.getElementById('widget-links');
+        ReactDOM.unmountComponentAtNode(widgetlink);
+    }
```

Comment:
can we standardize on whether we use `handleEvent` or `onEvent` naming for event handling functions? ([AirBnb's style guide](https://github.com/airbnb/javascript/tree/master/react#methods) implies using `onWhatever`, plus, it's shorter)